src/niamoto/core/plugins/transformers/class_objects/categories_mapper.py

```
@@ -0,0 +1,173 @@
+
+                result[category] = category_values
+
+            return result
+
+        except Exception as e:
+            if isinstance(e, DataTransformError):
+                raise e
+            raise DataTransformError(
+                "Failed to map categories",
+                details={"error": str(e), "config": config},
+            )
```

Ruff (0.8.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)